cmd/ndm_daemonset/probe/deletehandler.go

```diff
@@ -38,24 +38,63 @@ func (pe *ProbeEvent) removeBlockDeviceFromHierarchyCache(bd blockdevice.BlockDe
 }
 
 // deleteBlockDevice marks the block device resource as inactive
+// The following cases are handled
+//	1. Device using legacy UUID
+//	2. Device using GPT UUID
+//	3. Device using partition table UUID (zfs localPV)
+//  4. Device using the partition table / fs uuid annotation
 func (pe *ProbeEvent) deleteBlockDevice(bd blockdevice.BlockDevice, bdAPIList *apis.BlockDeviceList) error {
 
 	if !pe.removeBlockDeviceFromHierarchyCache(bd) {
 		return nil
 	}
 
-	uuid, ok := generateUUID(bd)
-	// this can happen if the device didn't have any unique identifiers
-	if !ok {
-		klog.Info("could not recreate UUID while removing device")
+	// try with gpt uuid
+	if uuid, ok := generateUUID(bd); ok {
+		existingBD := pe.Controller.GetExistingBlockDeviceResource(bdAPIList, uuid)
+		if existingBD != nil {
+			pe.Controller.DeactivateBlockDevice(*existingBD)
+			klog.V(4).Infof("deactivated device: %s, using GPT UUID", bd.DevPath)
+			return nil
+		}
+		// uuid could be generated, but the disk may be using the legacy scheme
+	}
+
+	// try with legacy uuid
+	legacyUUID, _ := generateLegacyUUID(bd)
```

Comment:
This code needs to move towards the end to handle a case like this:
- ndm 0.7 created a BD using legacyuuid for the devices used by zfs local pv
- upgrade to 0.8, a  new BD will be created with partition id
- when the device gets disconnected, only legacyuuid based device will be marked as inactive (noop)
- the new uuid based BD will remain active.

Reply:
Done.